### PR TITLE
complete the tests

### PR DESCRIPTION
**Pull request by peter279k**
# Changed log
- Complete the tests

### CODE REVIEW

File: tests/TagTest.php [hunk 1]
```diff
@@ -44,4 +44,152 @@ public function testVoidTag()
...
+    }
+
+    public function testRemoveAttribute()
+    {
+        $tag = new Tag('p');
+        $tag->setAttributes([
+            'class' => 'class_name'
+        ]);
+
+        $this->assertInstanceOf(Tag::class, $tag->removeAttribute('class'));
```
Comment (BenMorel) [inline tests/TagTest.php]: You could actually use `assertSame($tag, ...` to check that the same instance is returned!

File: tests/TagTest.php [hunk 2]
```diff
@@ -44,4 +44,152 @@ public function testVoidTag()
...
+    {
+        $tag = new Tag('img');
+        $tag->appendTextContent('this_is_appended_text_content');
+    }
+
+    public function testAppendTextContent()
+    {
+        $tag = new Tag('p');
+        $appendTextContent = 'this_is_appended_text_content';
+        $tag->appendTextContent($appendTextContent);
```
Comment (BenMorel) [inline tests/TagTest.php]: Can you check the return value here as well? Using `assertSame($tag, ...`

File: tests/TagTest.php [hunk 3]
```diff
@@ -44,4 +44,152 @@ public function testVoidTag()
...
+    {
+        $tag = new Tag('img');
+        $tag->appendHtmlContent('<span>this_is_appended_html_content</span>');
+    }
+
+    public function testAppendHtmlContent()
+    {
+        $tag = new Tag('p');
+        $appendHtmlContent = '<span>this_is_appended_text_content</span>';
+        $tag->appendHtmlContent($appendHtmlContent);
```
Comment (BenMorel) [inline tests/TagTest.php]: Same here

File: tests/TagTest.php [hunk 4]
```diff
@@ -44,4 +44,152 @@ public function testVoidTag()
...
+    {
+        $tag = new Tag('img');
+        $tag->append(new Tag('img'));
+    }
+
+    public function testIsEmptyShouldReturnTrue()
+    {
+        $tag = new Tag('p');
+        $tag->appendTextContent('append_text_content');
+        $tag->empty();
```
Comment (BenMorel) [inline tests/TagTest.php]: Please check the return value here as well

File: tests/TagTest.php [hunk 5]
```diff
@@ -44,4 +44,152 @@ public function testVoidTag()
...
+        $tag = new Tag('img');
+        $tag->renderClosingTag();
+    }
+
+    public function testTagInstanceShouldReturnRenderedString()
+    {
+        $tag = new Tag('p');
+        $appendTextContent = 'append_text_content';
+        $tag->appendTextContent($appendTextContent);
+        $tag->render();
```
Comment (BenMorel) [inline tests/TagTest.php]: You don't need to call `render()` before casting to string! `__toString()` returns `render()`.
You should actually return the return value of both `render()` and `(string) $tag` in this method.

File: tests/TagTest.php [hunk 6]
```diff
@@ -44,4 +44,152 @@ public function testVoidTag()
...
+
+        $this->assertSame('<p>' . $appendTextContent . '</p>', (string)$tag);
```
Comment (BenMorel) [inline tests/TagTest.php]: CS: please add a space after `(string)`

**Comment by peter279k** [comment]
Hi @BenMorel, thank you for your reply.
I jsut modify them for the request changes.

If having any issues, please feel free to send the comments!

**Comment by BenMorel** [comment]
Looks good, thanks! 👍

**Pull request merged**